src/core/collision.cpp

```diff
@@ -611,27 +657,61 @@ void handle_collisions() {
           p1->p.rotation = ROTATION_X | ROTATION_Y | ROTATION_Z;
           p2->p.rotation = ROTATION_X | ROTATION_Y | ROTATION_Z;
 
-          // The vs placement is done by the node on which p1 is non-ghost
-          if (!p1->l.ghost) {
-            bind_at_point_of_collision_calc_vs_pos(p1, p2, pos1, pos2);
+          // Positions of the virtual sites
+          bind_at_point_of_collision_calc_vs_pos(p1, p2, pos1, pos2);
+
+          // place virtual sites on the node where the base particle is not a
+          // ghost
+          if (!p1->l.ghost)
```

Comment:
Is it safe to deref p1 here (and p2 in the following)?

There is the corner case that one of the two particle pointers is null and the other is local which is not caught by the if. This should not happen as the two particles got queued by a short_range_loop? Should we at least comment on that fact?